src/modules/previewpane/powerpreview/dllmain.cpp

```diff
@@ -0,0 +1,55 @@
+#include "pch.h"
+#include <interface/powertoy_module_interface.h>
+#include "trace.h"
+#include "powerpreview.h"
+#include "CLSID.h"
+
+// Logic to shim the Activation of .Net Assembly by calling CoGetClassObject. CLSID's of Preview Handlers should be present in the registry.dat under /Classes/CLSID/{guid}.
+// See the existing Preview Handlers registry entry in registry.reg file.
+// This is required since MSIX currently not support .Net Assembly for Com Activation for Preview Handlers.
```

Comment:
IIUC, we need registry.dat/registry.reg and this library only to be able to place ourselves in the context where Windows has mapped our virtual registry w/ .NET COM classes registration endpoints to the "global registry view", so we can use `CoGetClassObject` to trigger Windows' COM class locator mechanism to load the actual .NET assembly. 

I've had a similar issue when implementing `notifications` as a C++/winrt library: it also had to have a these DLL functions exported. Still, it would be nice to avoid having those registry files. There's an option to have a so-called [mixed mode assembly](https://docs.microsoft.com/en-us/cpp/dotnet/mixed-native-and-managed-assemblies?view=vs-2019), which has both native C++ and C++/CLI parts. I've personally never used them, but maybe it'd be worth a try researching if it's possible to export those required native-DLL functions and at the same time returning the actual MarkdownPreviewHandler/etc. objects?

Reply:
MSIX doesn't support .Net Assemblies for `ComSurrogateServer` . For surrogate server MSIX require a path to a dll which have implementation of COM inproc server(DLLGetClassObject Export). For managed COM classes this implementation is in `mscoree.dll` . What is happening here the call is made to native dll first from MSIX perspective and in the native DLLGetClassObject implementation i'm calling the `CoGetClassObject` to activate the actual managed Com class. `Registry.dat` have the class registration key for .net assembly so that the native dll can activate it. 

To my understanding even if we use Mixed mode assembly and export the implementation of `DLLGetClassObject` we will need to provide a similar implementation as `mscoree.dll` and also initialize the CLR not sure if it's gonna be straight forward and need more investigation on what are the implementation details to achieve this. Not sure if this is in  the scope of the PR. We can create a separate issue to investigate if it's possible. Let me know what you think.

Reply:
I'd assume that Mixed mode assembly will handle CLR initialization for us, since C++/CLI was created to serve as a glue layer between native and managed. It might be not that straightforward in practice, of course, so I'm okay with creating a separate issue.